app/components/agent/Show.js

```diff
@@ -34,11 +35,22 @@ class AgentShow extends React.Component {
   };
 
   componentDidMount() {
-    this._agentRefreshInterval = setInterval(this.fetchUpdatedData, 5::seconds);
+    // Only bother setting up the delayed load and refresher if we've got an
+    // actual agent to play with.
+    if(this.props.agent && this.props.agent.id) {
+      this._agentRefreshInterval = setInterval(this.fetchUpdatedData, 5::seconds);
```

Comment:
I was gonna, but then `this.props.relay.setVariables({ isMounted: true })` needs to be in the `if` as well, so I figured this'd be the best place.